### PR TITLE
Add specs and requirements tables to API pages

### LINKED ISSUE #126: API pages need a Requirements section
Currently, we have the `HTMLMapmlViewerElement`, `HTMLLayerElement `and `HTMLFeatureElement `pages with no Requirements sections.  It would be ideal to include a tabulation of Requirements etc. when possible. especially where the API does not have a declarative equivalent, e.g. `HTMLMapmlViewerElement.geojson2mapml`, etc.

### PR DESCRIPTION
**Pull request by prushforth**
- [x] HTMLMapmlViewerElement
- [x] HTMLLayerElement
- [x] HTMLFeatureElement
- [x]  Custom Projections API - move into HTMLMapmlViewerElement page
- [x]  Custom Rendering API - rename to map-link-api / HTMLLinkElement page, add specs and requirements section

Closes #126

### CODE REVIEW

File: docs/api/layer-api.md [hunk 1]
```diff
@@ -261,4 +261,32 @@ window.onload = (event) => {
...
+<span class="discussion">under discussion</span></b></p>
+
+|  | Spec | Viewer | API |
+|:---------------------------------------------------------------------------------|:------: |:-----: |:---: |
+| [**Properties**](#properties) | full | full | full |
+| [**Methods**](#methods) | full | full | full |
+
+---
+
+> - [Edit this page on **Github**](https://github.com/Maps4HTML/web-map-doc/edit/main/docs/api/mapml-viewer-api.mdx)
```
Comment (AliyanH) [inline docs/api/layer-api.md:291]:
```suggestion
> - [Edit this page on **Github**](https://github.com/Maps4HTML/web-map-doc/edit/main/docs/api/layer-api.md)
```

**Review by AliyanH — APPROVED** [review]
LGTM, really nice to have it this organized!

Just noticed a small change you did to the English doc but missed for fr (DOMTokenList)
https://github.com/Maps4HTML/web-map-doc/blob/d3e813467c20aaf9762720157cecd855330ceb43/i18n/fr/docusaurus-plugin-content-docs/current/api/mapml-viewer-api.mdx#L42

**Comment by prushforth** [comment]
Sorry, it's a big enough change that you should check it out.  The custom rendering and custom projections descriptions need work, but that can wait until further work on those APIs happens, I think.

**Review by AliyanH — APPROVED** [review]
LGTM, the API section looks and makes sense now!

**Pull request merged**